sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamIOSourceRel.java

```diff
@@ -109,10 +137,11 @@ public RelOptCost computeSelfCost(RelOptPlanner planner, RelMetadataQuery mq) {
   @Override
   public BeamCostModel beamComputeSelfCost(RelOptPlanner planner, RelMetadataQuery mq) {
     NodeStats estimates = BeamSqlRelUtils.getNodeStats(this, mq);
-    return BeamCostModel.FACTORY.makeCost(estimates.getRowCount(), estimates.getRate());
+    return BeamCostModel.FACTORY.makeCost(
+        estimates.getRowCount() * getRowType().getFieldCount(), estimates.getRate());
   }
```

Comment:
Is this the most optimal way to ensure that the table with projects pushed-down is favored?

Reply:
Updated to multiplying total cost by `getRowType().getFieldCount()` stead of just `getRowCount`.